modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/index.js

```diff
@@ -148,14 +149,96 @@ export async function loadData(
 	}
 
 	if (sorts.length) {
-		url.searchParams.delete('sort');
-
-		url.searchParams.append(
+		url.searchParams.set(
 			'sort',
 			sorts.map((item) => `${item.key}:${item.direction}`).join(',')
 		);
 	}
 
+	if (Liferay.FeatureFlags['LPD-25230'] && additionalAPIURLParameters) {
+		const additionalAPIURLParametersArray =
+			additionalAPIURLParameters.split('&');
+
+		additionalAPIURLParametersArray.forEach((parameter) => {
+			const [key, value] = parameter.split('=');
+
+			const existingFilter = url.searchParams.get('filter');
+
+			if (key === 'filter' && existingFilter) {
+				url.searchParams.set(
+					'filter',
+					existingFilter + ' and ' + value
+				);
+			}
```

Comment:
We discussed this in a meet and the conclusion is that yes, there is a product need for the "merge" behavior. My concern is that this `' and '` may not be sufficient, because we have few different ways in building odata. See [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/controls/filters/implementation/SelectionFilter.tsx#L104-L139) and [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/controls/filters/implementation/DateRangeFilter.tsx#L82-L107):
- Collection entity field has strict format, enclosed in parentheses after keyword `any`, `/any(x:(x eq 1) and (x eq 2))`. I am not sure that appending `and` will work as expected. It might just break parsing, or have no effect. 
- Should we enclose `existingFilter` and `value` with parentheses? What would happen if exclude `or` is in `existingFilter`? We are at a mercy of order of operators.
- Client extension filter can have any format of odata, but I think we can safely ignore that case. There is no way to cover it.

Can we test all [entity field types](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-api/src/main/java/com/liferay/frontend/data/set/constants/FDSEntityFieldTypes.java)? Custom FDS module has example of all 3/4 entity field types, missing only `date-time`. To test if they work, I think you can temporarily hardcode to always add some additional parameters.

Reply:
I agree that we can enclose with parentheses to keep the order of operators with a more expected outcome.

With a collection entity field, I'd have to figure out how to test this, but I'd assume if the `existingFilter` was [a lambda operator](https://help.liferay.com/hc/en-us/articles/360036343152-Filter-Sort-and-Search#lambda-operators) `keywords/any(k:contains(k,'substring1'))` and the new `value` was also similar `keywords/any(k:contains(k,'substring2')`, it should be fine combined as `(keywords/any(k:contains(k,'substring1'))) and (keywords/any(k:contains(k,'substring2')))`

I'm unclear what you mean by testing all entity field types? I think it'd be up to the user to format the value that'd be added on.

Reply:
I meant to manually test all entity field types, as a way to manually test all types of odata formats we have. But, it looks like we have it all covered, other entity types look to me like they do fit `(existing) and (new)`. If there is some odd edge case, we'll see it down the line. So, as far as I'm concerned, feel free to ship this, after you add the parenthesis.